src/modules/components/proposal/proposalActions/proposalActionsContainer/proposalActionsContainer.tsx

```diff
@@ -0,0 +1,94 @@
+import classNames from 'classnames';
+import { cloneElement, isValidElement, useState, type ReactElement } from 'react';
+import { Accordion, Button, Card, Heading, type IAccordionContainerBaseProps } from '../../../../../core';
+import { useOdsModulesContext } from '../../../odsModulesProvider';
+import { type IProposalActionsActionProps } from '../proposalActionsAction';
+
+export interface IProposalActionsContainerProps extends Omit<IAccordionContainerBaseProps<true>, 'isMulti'> {
+    /**
+     * Actions as custom children to render
+     */
+    children?: ReactElement<IProposalActionsActionProps> | Array<ReactElement<IProposalActionsActionProps>>;
+    /**
+     * Additional class names for styling
+     */
+    className?: string;
+}
+
+export const ProposalActionsContainer: React.FC<IProposalActionsContainerProps> = (props) => {
+    const { className, children } = props;
+    const [expandedItems, setExpandedItems] = useState<string[]>([]);
+    const { copy } = useOdsModulesContext();
+
+    const childrenCount = Array.isArray(children) ? children.length : children ? 1 : 0;
+
+    const handleToggleAll = () => {
+        if (expandedItems.length === childrenCount) {
+            setExpandedItems([]);
+        } else {
+            setExpandedItems(Array.from({ length: childrenCount }, (_, index) => `${index}-action`));
```

Comment:
I'd set the index to the actions simply to `index` to avoid having to build the action id every time